addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/channel/AbstractSpec.java

```diff
@@ -43,9 +43,13 @@ protected AbstractSpec(@Nullable ChannelConfiguration channelConfiguration, Stri
      * @param ga the group address configuration
      * @return a group address object
      */
-    protected final GroupAddress toGroupAddress(GroupAddressConfiguration ga) {
+    protected final GroupAddress toGroupAddress(@Nullable GroupAddressConfiguration ga) {
         try {
-            return new GroupAddress(ga.getGA());
+            if (ga != null) {
+                return new GroupAddress(ga.getGA());
+            } else {
+                throw new IllegalArgumentException("GroupAddressConfiguration should contain a group address");
```

Comment:
Why did you change the signature to be `@Nullable` and then throw a IAE in case `null` is passed?

Reply:
I should revert this and do the null check before calling this. Underlying to this is that in the advanced binding there can be ChannelConfigurations that do not have a MainGA (toGroupAddress is called in ReadResponseSpecImpl)